### ISSUE TITLE
Save/Bookmark buttons in feed do not have unique accessible names

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#18286]
**Describe the bug**

In the home feed, tags feed, search results, and a user's profile page, we display feed cards including a 'bookmark' button to save to the reading list. The accessible name of all of these buttons is "Save".

This means that a screen reader user will hear "Save / Save / Save" many times without context of which post the button saves.

**To Reproduce**

- Go to the home feed
- Try with e.g. VoiceOver opening the rotor and viewing the controls list
- Notice you have a huge number of buttons in the list that just say "Save" with no clue as to what they are saving

**Expected behavior**

- Each button should have a unique label, e.g. including the post's title

**Comment by Link2Twenty** [comment]
This is the component that renders the button https://github.com/forem/forem/blob/main/app/javascript/articles/components/SaveButton.jsx

It looks like the button already takes the whole article object so there's plenty of data to make an aria label from
https://github.com/forem/forem/blob/main/app/javascript/common-prop-types/article-prop-types.js

It might be worth changing `SaveButton` to a Functional component, rather than a Class based one, though perhaps that is for another issue.

**Comment by aitchiss** [comment]
Worth bearing in mind that the `SaveButton` component is only used for the logged-in feed, and only the first page of results 🙃 

We'll need to keep things consistent between that component and also
- `buildArticleHTML.js` and
- `_single_story.html.erb`

**Comment by Link2Twenty** [comment]
Oh, goodness. Might be worth unifying that at some point 😅.

**Comment by aitchiss** [comment]
> Oh, goodness. Might be worth unifying that at some point 😅.

I know I know it sounds like a no-brainer but there's a lot going on there 😅 One day...!

**Comment by eevajonnapanula** [comment]
If no one is working on this, I'd like to take it 🙋🏻‍♀️
